terraform.tf

```diff
@@ -1,9 +1,9 @@
 terraform {
-  required_version = "~> 0.12.0"
+  required_version = ">= 0.12.0, < 0.14.0"
 }
 
 provider "aws" {
-  version = "~> 2.7"
+  version = ">= 2.7.0, <= 3.11.0"
```

Comment:
I think I want to be less strict here because 3.12.0 will be out shortly I'm sure and this will block users from using it.  I would say let's just put `>= 2.7.0`